File: DuDoong-Common/src/main/java/band/gosrock/common/validator/EnumValidator.java
package band.gosrock.common.validator;


import band.gosrock.common.annotation.Enum;
import java.util.Arrays;
import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

public class EnumValidator implements ConstraintValidator<Enum, java.lang.Enum> {
    @Override
    public boolean isValid(java.lang.Enum value, ConstraintValidatorContext context) {
        if (value == null) return false; // null 값 허용 여부
        Class<?> reflectionEnumClass = value.getDeclaringClass();
        return Arrays.asList(reflectionEnumClass.getEnumConstants()).contains(value);
    }
}

File: DuDoong-Common/src/main/kotlin/band/gosrock/common/validator/EnumValidator.kt
package band.gosrock.common.validator

import band.gosrock.common.annotation.Enum as EnumAnnotation
import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class EnumValidator : ConstraintValidator<EnumAnnotation, kotlin.Enum<*>> {
    override fun isValid(value: kotlin.Enum<*>?, context: ConstraintValidatorContext): Boolean {
        if (value == null) return false
        val reflectionEnumClass = value.declaringJavaClass
        return reflectionEnumClass.enumConstants.contains(value)
    }
}
